model/fv_dynamics.F90

```diff
@@ -1097,16 +1198,24 @@ end subroutine Rayleigh_fast
 
 
 
- subroutine Rayleigh_Super(dt, npx, npy, npz, ks, pm, phis, tau, u, v, w, pt,  &
+ subroutine Rayleigh_Super(dt, npx, npy, npz, ks, pm, phis, tau, tau_w, u, v, w, pt,  &
+#ifdef MULTI_GASES
+                           q, ncnst,  &
+#endif
                            ua, va, delz, agrid, cp, rg, ptop, hydrostatic,     &
                            conserve, rf_cutoff, gridstruct, domain, bd)
     real, intent(in):: dt
     real, intent(in):: tau              !< time scale (days)
+    real, intent(in):: tau_w            !< time scale (days) for w 
     real, intent(in):: cp, rg, ptop, rf_cutoff
     real, intent(in),  dimension(npz):: pm
     integer, intent(in):: npx, npy, npz, ks
     logical, intent(in):: hydrostatic
+#ifdef MOLECULAR_DIFFUSION
+    real, intent(in):: conserve
+#else
     logical, intent(in):: conserve
+#endif
```

Comment:
Changing the type of a variable in this manner and using it below is dangerous.  Especially since you are using a completely different variable when calling the routine.